agent/balrogagent/cmd.py

```diff
@@ -0,0 +1,70 @@
+import aiohttp
+import asyncio
+import logging
+import time
+import traceback
+
+from . import client
+from .changes import get_telemetry_uptake, is_ready
+from .log import configure_logging
+
+
+async def run_agent(loop, balrog_api_root, balrog_username, balrog_password, telemetry_api_root, sleeptime=30):
+    auth = aiohttp.BasicAuth(balrog_username, balrog_password)
+
+    while True:
+        try:
+            async with aiohttp.ClientSession(auth=auth, loop=loop) as session:
+                logging.debug("Looking for active scheduled changes...")
+                # TODO: sometimes getting "unclosed response" and "unclosed connection" errors.
+                resp = await client.request(session, balrog_api_root, "/csrf_token", method="HEAD")
+                csrf_token = resp.headers["X-CSRF-Token"]
```

Comment:
I think the reason I did this was because it seemed dirty to modify "data" in request() to add the CSRF token. It doesn't seem like such a strong reason now though...what do you think @rail?